### PR TITLE
Clarify case normalization in db spec

### PR DESCRIPTION
**Pull request by gregkalapos**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/886

## Changes

Adding note about case normalization to following attributes:
- db.collection.name
- db.namespace
- db.operation.name

The added note suggested in  https://github.com/open-telemetry/semantic-conventions/issues/886
> It is RECOMMENDED to capture the value as provided by the application without attempting to do any case normalization.

## Merge requirement checklist

* [x] CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
I think probably needs a change log

**Review by lmolkova — APPROVED** [review]

File: model/registry/db.yaml
Comment (lmolkova) [inline model/registry/db.yaml:1]: please add a changelog.
(Adding a comment so it's not accidentally merged)

Comment (gregkalapos) [reply model/registry/db.yaml:1]: 👍 added changelog

**Comment by gregkalapos** [comment]
CI fails with:

```
34s
Run make markdown-link-check
  make markdown-link-check
  shell: /usr/bin/bash -e {0}
fatal: No names found, cannot describe anything.
semantic-conventions@ /home/runner/work/semantic-conventions/semantic-conventions
└── markdown-link-check@3.11.2


  ERROR: 1 dead links found in ./docs/attributes-registry/db.md !
  [✖] https://docs.datastax.com/en/cassandra-oss/3.0/cassandra/dml/dmlConfigConsistency.html → Status: 404
make: *** [Makefile:47: markdown-link-check] Error 1
Error: Process completed with exit code 2.
```

That link works from my browser - and the PR doesn't really change anything around it. Strange. Trying to figure this out.

**Pull request merged**